### DISCUSSION TITLE
[Request] /cmi kill command

### DISCUSSION THREAD

**Discussion by TheWranger** [Zrips/CMI#490] [Feature Requests]
Add a /cmi kill command where we can define what the death cause will be. /cmi suicide should be kept separate.

**Comment by CloudeLecaw** [comment]
a bit mire specifiy for a request would be cool. like
**cmi kill [suicide/Player] (reason)** ? 
permission: **cmi.command.kill**  and **cmi.command.kill.others**
   so u have Option for suicide and kill others
also add an -s Option for doing it silent.. not every Player must know an admin or console killed him^^

if No reason is set up then a Default reason (config) is choosen
if no suicide or Playername is choosen then a Default Option (config?) should be choosen
also thought abotu respecting Nicked Players ;)

**Comment by TheWranger** [comment]
There's no need to add a suicide feature to /CMI kill.
The death cause is the internal spigot death cause, not a reason for the player dying.
I imagine this would be better used for debugging purposes.

**Comment by CloudeLecaw** [comment]
i see that it may iss more simply to use spigot Default death reasons...may make sense for analytiscs or what after.
and no it does not make sense to Keep suicide seperate. /kill without Argument kills you and done
so you can kill yourself and others (with extra permision) and if an admin Forgets to use Arguments ist his own fault if he kill hisself. (and may make some plasyers get a smile on their face :D)

There is no reason for an extra suicide command
Because ist in essentials? is not really an Argument for an extra command.

**Comment by TheWranger** [comment]
I disagree, we need too commands:
/suicide (what we currently have)
/kill [PLAYER] [SPIGOT_DEATHCAUSE]

/kill should be used for killing people with cusom causes, /suicide should be left alone.

**Comment by CloudeLecaw** [comment]
**/cmi suicide (playerName) (-s) - Kill your self or target player** 
permisisons:
**cmi,command.suicide**   and  **cmi.command.suicide.others**.


i see the kill Option for suicide and others is done allready and a kill alais u can create by yourself (CustomAlias)

i see no reason for adding the DeathReason. Why should you? For printing out a faked DeathReason when you killed someone?

pls close this page, because the wanted Option is allready there.

if you want faked DeathReasons for killing someone then you also can creat CustomAlias for this.
li - 1. command kill him 2. command print him a meassge why he died…. 
**example:**
```
  killbuttom:
  - cmi suicide $1 -s
  - cmi broadcast $1 hit the button a bit to hard.
```
and if u want Spigot causes then print the Right cause in seocund line.
here you find [possible Minecraft causes of death](http://minecraft.wikia.com/wiki/Death)

"but this command can be used by everone?"
On your Server everyone can broadcast and kill.others?
interesting…
but ok..
then work with perm:cmi.command.customalias.killbuttom! if u want an extra permission ;)
```
  killbuttom:
  - perm:cmi.command.customalias.killbuttom! cmi suicide $1 -s
  - perm:cmi.command.customalias.killbuttom! cmi broadcast $1 hit the button a bit to hard.

``` 
renember your own worls ;) 

> Don't really see the point of this, seems like just adding a feature because Essentials has it.

so pls close this issues :)

**Comment by TheWranger** [comment]
1. There are several good reasons for setting a custom death cause, mainly being that it's really useful for debugging and also changing what death message is displayed. What you are suggesting is that people instead spoof a death message which would be inefficient and cause issues with plugins that handle death messages.
2. Here's spigots list of internal death/damage causes: https://hub.spigotmc.org/stash/projects/SPIGOT/repos/bukkit/browse/src/main/java/org/bukkit/event/entity/EntityDamageEvent.java#280
3. Esssentials doesn't allow you to change what death cause /kill uses.

**Comment by CloudeLecaw** [comment]
> Esssentials doesn't allow you to change what death cause /kill uses.

jep because ist not nessesary


changing what death message is displayed is the Job of other deathmeassge Plugins. your Debugging Thing is only sensfull in a simulated death meassage and never on a real deth, real death killis with the spigot reasons would be unnesseary and inefficent for Debugging. then you also could kill yourself manuelly each time ...this would be inefficent to...Well

i still think this is not nessesary..but before we now turn on with saying it is, it is not.. lets dev decide if ist a nessesary feature for CMI or not :P
